core/domain/story_jobs_one_off_test.py

```diff
@@ -143,7 +143,7 @@ def test_migration_job_converts_old_story(self):
         topic_services.add_canonical_story(
             self.albert_id, self.TOPIC_ID, self.STORY_ID)
         story = (
-            story_fetchers.get_story_by_id(self.STORY_ID))
+            story_models.StoryModel.get(self.STORY_ID))
```

Comment:
What's the reason for this change? If you read the description of the test I think the intended result is for the story to end up as v2, so the previous code should be correct?

Reply:
Hi @seanlip, the `get_story_by_id` method calls `get_story_from_model` which does the migration but if you look at the code below we are asserting the schema version to still point to the older value and then we do a migration following which we assert the new value for schema version.

Ditto for Topic below.

Reply:
Oh, I see -- sorry, my bad. Thanks! But then you need to do the same thing below, right -- i.e. use StoryModel.get()?

Perhaps it might be a good idea to do both calls in both the before/after places, and assert that both values are correct before and after the job is run.

Reply:
Regarding [this comment](https://github.com/oppia/oppia/pull/8498#discussion_r379891970):

>Perhaps it might be a good idea to do both calls in both the before/after places, and assert that both values are correct before and after the job is run.

I'm not sure I understand, do you mean have both (1) story_fetchers.get_story_by_id() and (2) story_models.StoryModel.get() before and after the migration and assert new schema value for get_story_by_id for (1) and old schema value for (2)
And after migration assert new schema value for both (1) and (2)?

If so, do you think we should have 2 separate tests one testing (1) and another testing (2)?

Reply:
I would suggest doing both in the same test, because I don't think the order should matter.

Are you sure the order matters? I wouldn't have thought so, because GET operations should not change backend state. I know that the things you're going to retrieve will end up being different but I don't think the GET operations will change the datastore (so swapping them with each other should not change the results of the test). Is that not the case? If the above doesn't hold, let's have a longer discussion about this because there are more serious issues here.

Reply:
Yes, you're right the order doesn't matter. I've added both in the same test now. Thanks!